index.d.ts

```diff
@@ -1584,6 +1584,29 @@ declare namespace Moleculer {
 			delimiter(value: string): void;
 		}
 	}
+
+	namespace Utils {
+		function isFunction(func: any): boolean;
+		function isString(str: any): boolean;
+		function isObject(obj: any): boolean;
+		function isPlainObject(obj: any): boolean;
+		function isDate(date: any): boolean;
+		function flatten(arr: any[]): any[];
+		function humanize(millis: number): string;
+		function generateToken(): string;
+		function removeFromArray(arr: any[], item: any): any[];
+		function getNodeID(): string;
+		function getIpList(): any[];
+		function isPromise(promise: any): boolean;
+		// function polyfillPromise(P: Promise): any; // From Utils -> NOT USED & NOT TESTED YET !!!
+		function clearRequireCache(filename: string): void;
+		function match(text: string, pattern: string): boolean;
+		function deprecate(prop: any, msg: string): void;
```

Comment:
@icebob Can you fill me in on what this function is used for?  I don't see it referenced anywhere nor is the `deprecateList` it mutates used anywhere.  I'm not sure what the inputs to this function should be, although the jsdoc indicates it as `Object|Function|string`.  Is there any real expectation of what the input to this should be?

Reply:
Thank you @shawnmcknight for your review. I'm then waiting from @icebob on this to update.

Reply:
Yeah, the input can be anything. You can use it to write a deprecate warning in the modules. E.g if there is a deprecated method in a class, and somebody call, that you can call it with the method. So this function will store the pointer of method and won't print the deprecate messages all times, just only one. But if there is a deprecated broker option, in this case, you can call it with a string. So normally it will be called with (Object,Function, String), but can be anything. 

Currently it's not used because there is no deprecation in the codebase, but it would be used in the future.